h2o-assemblies/main/build.gradle

```diff
@@ -92,8 +92,57 @@ task copyJar(type: Copy) {
     rename { it.replace(assembly, allInOne) }
 }
 
+downloadLicenses {
+    ext.apacheTwo = license('Apache License, Version 2.0', 'http://opensource.org/licenses/Apache-2.0')
+    ext.bsd = license('BSD License', 'http://www.opensource.org/licenses/bsd-license.php')
+
+    licenses = [
+            'commons-beanutils:commons-beanutils:1.7.0': apacheTwo,
+            'asm:asm-parent:3.1': bsd,
+            'org.codehaus.jettison:jettison:1.1': apacheTwo,
+            'javax.servlet.jsp:jsp-api:2.1': license('CDDL + GPLv2 with classpath exception', 'https://glassfish.dev.java.net/nonav/public/CDDL+GPL.html'),
+            'org.apache.zookeeper:zookeeper:3.4.6': apacheTwo
+    ]
+
+    aliases = [
+            (apacheTwo) : ['The Apache Software License, Version 2.0', 'Apache 2', 'Apache License Version 2.0', 'Apache License, Version 2.0', 'Apache License 2.0', license('Apache License', 'http://www.apache.org/licenses/LICENSE-2.0')],
+            (bsd) : ['BSD', license('New BSD License', 'http://www.opensource.org/licenses/bsd-license.php')]
+    ]
+}
+
+task licenseReport(dependsOn: project.tasks.getByName('downloadLicenses')) {
+    doLast {
+        def jsonLicensesFile = file("${buildDir}/reports/license/dependency-license.json")
+        def csvLicensesFile = file("${buildDir}/reports/license/dependency-license.csv")
+        def licenses = new groovy.json.JsonSlurper().parseText(jsonLicensesFile.text)
+        csvLicensesFile.withWriter('UTF-8') { w ->
+            boolean xgboostDefined = false
+            w.write('"Dependency","File","License Name","License URL"\n')
+            licenses.dependencies.each { dep ->
+                dep.licenses.each {
+                    xgboostDefined |= dep.file.startsWith("xgboost4j")
+                    boolean licenseValid = it.name != "No license found"
+                    if (! licenseValid) {
+                        if (dep.name == "asm:asm-parent:3.1") {
+                            // FIXME: manual override, this doesn't work for some reason!
```

Comment:
?

Reply:
Something is not right with the plugin. I have "asm:asm-parent:3.1" defined in lincenses but it still generates "No license found" in the JSON file. Not sure what is wrong there.

This is to make sure that we do have the license information in the final output.